velox/docs/functions/spark/array.rst

```diff
@@ -196,6 +196,21 @@ Array Functions
         SELECT size(NULL, true); -- -1
         SELECT size(NULL, false); -- NULL
 
+.. spark:function:: slice(array(E), start, length) -> array(E)
+
+    Returns a subarray starting at 1-based index ``start`` or from end if negative, with ``length`` elements.
+    Returns elements between ``start`` and the end of the array if ``start + length`` is outside of the array.
+    Returns empty array if ``start`` point outside of the array or ``length`` is 0.
+    Throws exception if ``start`` is 0 or ``length`` is negative. ::
```

Comment:
Is the behavior exactly the same with Presto's slice function except for the type of start and length? If not, have we added tests to cover these specific cases?

Reply:
Yes, they're same, the spark slice function is orginally from Presto's, can see https://issues.apache.org/jira/browse/SPARK-23930.

Also, I checked it when I tried to support it inside Spark. And I run these cases inside spark and confirm the behavior is the same as the spark.

Reply:
```java
spark-sql> SELECT slice(array(1, 2, 3, 4), 2, 2);
[2,3]
Time taken: 5.316 seconds, Fetched 1 row(s)
spark-sql> SELECT slice(array(1, 2, 3, 4), -2, 2);
[3,4]
Time taken: 0.089 seconds, Fetched 1 row(s)
spark-sql> SELECT slice(array(1, 2, 3, 4), 5, 1);
[]
Time taken: 0.065 seconds, Fetched 1 row(s)
spark-sql> SELECT slice(array(1, 2, 3, 4), 2, 5);
[2,3,4]
Time taken: 0.071 seconds, Fetched 1 row(s)
spark-sql> SELECT slice(array(1, 2, 3, 4), 2, 0);
[]
Time taken: 0.06 seconds, Fetched 1 row(s)
spark-sql> SELECT slice(array(1, 2, 3, 4), 1, -1);
java.lang.RuntimeException: Unexpected value for length in function slice: length must be greater than or equal to 0.
spark-sql> SELECT slice(array(1, 2, 3, 4), 0, 1);
java.lang.RuntimeException: Unexpected value for start in function slice: SQL array indices start at 1.
```